core/network/impl/router_libp2p.cpp

```diff
@@ -76,27 +80,31 @@ namespace kagome::network {
   }
 
   bool RouterLibp2p::prepare() {
-    app_state_manager_->takeControl(*block_announce_protocol_.get());
-    app_state_manager_->takeControl(*grandpa_protocol_.get());
-
-    app_state_manager_->takeControl(*sync_protocol_.get());
-    app_state_manager_->takeControl(*state_protocol_.get());
-    app_state_manager_->takeControl(*warp_protocol_.get());
-    app_state_manager_->takeControl(*beefy_protocol_.get());
-    app_state_manager_->takeControl(*beefy_justifications_protocol_.get());
-    app_state_manager_->takeControl(*light_protocol_.get());
-
-    app_state_manager_->takeControl(*propagate_transactions_protocol_.get());
-
-    app_state_manager_->takeControl(*collation_protocol_.get());
-    app_state_manager_->takeControl(*validation_protocol_.get());
-    app_state_manager_->takeControl(*req_collation_protocol_.get());
-    app_state_manager_->takeControl(*req_pov_protocol_.get());
-    app_state_manager_->takeControl(*fetch_chunk_protocol_.get());
-    app_state_manager_->takeControl(*fetch_available_data_protocol_.get());
-    app_state_manager_->takeControl(*statement_fetching_protocol_.get());
-
-    app_state_manager_->takeControl(*send_dispute_protocol_.get());
+    main_pool_handler_->execute([wp = weak_from_this()] {
```

Comment:
missing `setProtocolHandler`
```suggestion
    REINVOKE(main_pool_handler_)
```